lib/puppet/provider/user/hpux.rb

```diff
@@ -19,14 +22,23 @@
     value !~ /\s/
   end
 
-  has_features :manages_homedir, :allows_duplicates, :manages_passwords
+  has_features :manages_homedir, :allows_duplicates, :manages_passwords, :manages_password_age, :manages_expiry
 
   def deletecmd
     super.insert(1,"-F")
   end
 
   def modifycmd(param,value)
-    super.insert(1,"-F")
+     cmd = super(param, value)
+     cmd << "-F"
+     if self.trusted == "Trusted"
+        cmd << ";"
```

Comment:
I could not find any code references on the best way to make it append 
the command to have
/usr/sbin/lbin/usermod.sam ... -F  --> Which changes the password hash 
for the user.

and, if trusted,

Add in a second command to reset the password aging to 0 days via 
/usr/lbin/modprpw -v -k <user name>.

This is the command I run manually when I am pushing a user password 
hash to a system and then resetting their expiration back to 0 days.

Suggestions on a better way to implement this routine?  I am still 
learning Ruby and trying to find the best way to get this code to 
properly change the passwords as this is part of the base Puppet code, 
versus having been a module.

Thanks!

On 1/2/2015 2:15 PM, Michael Smith wrote:
>
> In lib/puppet/provider/user/hpux.rb 
> <https://github.com/puppetlabs/puppet/pull/3443#discussion-diff-22419246>:
>
> >
> >    def deletecmd
> >      super.insert(1,"-F")
> >    end
> >
> >    def modifycmd(param,value)
> > -    super.insert(1,"-F")
> > +     cmd = super(param, value)
> > +     cmd << "-F"
> > +     if self.trusted == "Trusted"
> > +        cmd << ";"
>
> What is this doing? Why use a semi-colon - which appears to just 
> ignore the 1st command - instead of just returning the modprpw call?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/puppetlabs/puppet/pull/3443/files#r22419246>.
>